common/static/common/js/discussion/utils.js

```diff
@@ -321,7 +321,8 @@
             imageUploadUrl = this.urlFor('upload');
             _processor = function(self) {
                 return function(text) {
-                    return self.postMathJaxProcessor(text);
+                    // HTML returned by Markdown is assumed to be safe to render
+                    return self.postMathJaxProcessor(edx.HtmlUtils.HTML(text)).toString();
```

Comment:
The wmd editor preview pane rendering logic expects a string, it doesn't know how to handle an HTMLSnippet. This is the last possible point at which we can do the HTMLSnippet -> string conversion before it gets passed off to some other rendering logic outside these files.